### PR TITLE
op-guide: add a guide about upgrading to 3.0

### PR DESCRIPTION
**Pull request by superlzs0476**
@liubo0127  PTAL Any need update?  -_-!

### CODE REVIEW

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+title: TiDB 3.0 升级操作指南
+category: deployment
+---
+
+# TiDB 3.0 升级操作指南
+
+本文档适用于从 TiDB 2.0 版本（v2.0.1 及之后版本）或 TiDB 2.1 RC 版本升级到 TiDB 3.0 GA 版本。TiDB 3.0 版本兼容 Kafka 版本的 TiDB-Binlog 以及 集群模式的 TiDB-Binlog 。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
本文档适用于从 TiDB 2.0 版本（v2.0.1 及之后版本）或 TiDB 2.1 RC 版本升级到 TiDB 3.0 GA 版本。TiDB 3.0 版本兼容 [Kafka 版本的 TiDB-Binlog](/tools/tidb-binlog-kafka.md) 以及[集群模式的 TiDB-Binlog](/tools/tidb-binlog-cluster.md)。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## 升级兼容性说明
+
+- loading
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: What does `loading` mean here?

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+## 注意事项
+
+- loading
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: ditto

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ansible 2.5.14
+$ pip show jinja2
+Name: Jinja2
+Version: 2.10
+$ pip show jmespath
+Name: jmespath
+Version: 0.9.0
+```
+
+> **注意**：请务必按以上文档安装 Ansible 及其依赖。确认 Jinja2 版本是否正确，否则启动 Grafana 时会报错。确认 jmespath 版本是否正确，否则滚动升级 TiKV 时会报错。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
> **注意：**
>
> 请务必按以上文档安装 Ansible 及其依赖。确认 Jinja2 版本是否正确，否则启动 Grafana 时会报错。确认 jmespath 版本是否正确，否则滚动升级 TiKV 时会报错。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+## 在中控机器上下载 TiDB-Ansible
+
+以 `tidb` 用户登录中控机并进入 `/home/tidb` 目录，备份 TiDB 2.0 版本或 TiDB 2.1 rc 版本的 tidb-ansible 文件夹：
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: Are the versions of tidb-ansible correct?

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+```
+$ mv tidb-ansible tidb-ansible-bak
+```
+
+下载最新 tidb-ansible release-2.1 分支，默认的文件夹名称为 `tidb-ansible`。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: Not release-3.0?

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+1. 请确认 `ansible_user` 配置的是普通用户。为统一权限管理，不再支持使用 root 用户远程安装。默认配置中使用 `tidb` 用户作为 SSH 远程用户及程序运行用户。
+
+    ```
+    ## Connection
+    # ssh via normal user
+    ansible_user = tidb
+    ```
+
+    可参考[如何配置 ssh 互信及 sudo 规则](../op-guide/ansible-deployment.md#在中控机上配置部署机器-ssh-互信及-sudo-规则) 自动配置主机间互信。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
    可参考[如何配置 ssh 互信及 sudo 规则](../op-guide/ansible-deployment.md#在中控机上配置部署机器-ssh-互信及-sudo-规则)自动配置主机间互信。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    # high-concurrency: 8
+    # normal-concurrency: 8
+    # low-concurrency: 8
+```
+
+单机多 TiKV 实例情况下，需要修改这三个参数，推荐设置：`实例数 * 参数值 = CPU 核数 * 0.8`。
+
+## 下载 TiDB 3.0 binary 到中控机
+
+确认 `tidb-ansible/inventory.ini` 文件中 `tidb_version = v3.0.0`，然后执行以下命令下载 TiDB 2.1 binary 到中控机。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: I think it should be 3.0?

```suggestion
确认 `tidb-ansible/inventory.ini` 文件中 `tidb_version = v3.0.0`，然后执行以下命令下载 TiDB 3.0 binary 到中控机。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+```
+$ ansible-playbook local_prepare.yml
+```
+
+## 滚动升级 TiDB 集群组件
+
+> **注意**：为优化 TiDB 集群组件的运维管理考虑，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行调整，注意滚动升级 TiDB 集群组件操作略有不同，注意升级前后 `process_supervision` 参数配置要保持一致。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: Please check whether this is what you mean.

```suggestion
> **注意：**
>
> 为优化 TiDB 集群组件的运维管理，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行了调整。与之前版本相比，滚动升级 TiDB 3.0 版本集群组件的操作略有不同，注意升级前后 `process_supervision` 参数配置须保持一致。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+如果 `process_supervision` 变量使用默认推荐 `systemd` 参数，则通过 excessive_rolling_update.yml 滚动升级 TiDB 集群。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
如果 `process_supervision` 变量使用默认推荐的 `systemd` 参数，则通过 `excessive_rolling_update.yml` 滚动升级 TiDB 集群。  
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+## 滚动升级 TiDB 集群组件
+
+> **注意**：为优化 TiDB 集群组件的运维管理考虑，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行调整，注意滚动升级 TiDB 集群组件操作略有不同，注意升级前后 `process_supervision` 参数配置要保持一致。
+
+如果 `process_supervision` 变量使用默认推荐 `systemd` 参数，则通过 excessive_rolling_update.yml 滚动升级 TiDB 集群。  
+```
+$ ansible-playbook excessive_rolling_update.yml
+```
+如果 `process_supervision` 变量使用默认推荐 `supervise` 参数，则通过 rolling_update.yml 滚动升级 TiDB 集群。
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
如果 `process_supervision` 变量使用默认推荐的 `supervise` 参数，则通过 `rolling_update.yml` 滚动升级 TiDB 集群。 
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+## 滚动升级 TiDB 集群组件
+
+> **注意**：为优化 TiDB 集群组件的运维管理考虑，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行调整，注意滚动升级 TiDB 集群组件操作略有不同，注意升级前后 `process_supervision` 参数配置要保持一致。
+
+如果 `process_supervision` 变量使用默认推荐 `systemd` 参数，则通过 excessive_rolling_update.yml 滚动升级 TiDB 集群。  
+```
+$ ansible-playbook excessive_rolling_update.yml
+```
```
Comment (lilin90) [inline op-guide/tidb-v3.0-upgrade-guide.md]: Please add a blank line both before and after the code block.

**Comment by lilin90** [comment]
@superlzs0476 Would you please address the comments? Thanks!

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 5 above)
Comment (liubo0127) [reply op-guide/tidb-v3.0-upgrade-guide.md]: "备份 TiDB 2.0 版本或 TiDB 2.1 版本的 tidb-ansible 文件夹"

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 2 above)
Comment (superlzs0476) [reply op-guide/tidb-v3.0-upgrade-guide.md]: I don't know who to write here or delete it.

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 3 above)
Comment (superlzs0476) [reply op-guide/tidb-v3.0-upgrade-guide.md]: I don't know who to write here or delete it.

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 5 above)
Comment (superlzs0476) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
以 `tidb` 用户登录中控机并进入 `/home/tidb` 目录，备份 TiDB 2.0 版本或 TiDB 2.1 rc 版本的 tidb-ansible 文件夹：
以 `tidb` 用户登录中控机并进入 `/home/tidb` 目录，备份 TiDB 2.0 版本或 TiDB 2.1 版本的 tidb-ansible 文件夹：
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+## 在中控机器上下载 TiDB-Ansible
+
+以 `tidb` 用户登录中控机并进入 `/home/tidb` 目录，备份 TiDB 2.0 版本或 TiDB 2.1 版本的 tidb-ansible 文件夹：
+
+```
+$ mv tidb-ansible tidb-ansible-bak
+```
+
+下载最新 tidb-ansible release-2.1 分支，默认的文件夹名称为 `tidb-ansible`。
```
Comment (superlzs0476) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
下载最新 tidb-ansible release-3.0 分支，默认的文件夹名称为 `tidb-ansible`。
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+
+```
+$ git clone -b release-2.1 https://github.com/pingcap/tidb-ansible.git
```
Comment (superlzs0476) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
$ git clone -b release-3.0 https://github.com/pingcap/tidb-ansible.git
```

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 13 above)
Comment (lilin90) [reply op-guide/tidb-v3.0-upgrade-guide.md]: @superlzs0476 I updated the link path on the above suggestion.

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 14 above)
Comment (lilin90) [reply op-guide/tidb-v3.0-upgrade-guide.md]: @liubo0127 PTAL

**Comment by lilin90** [comment]
@liubo0127 Could you please take a review and leave comments where @superlzs0476 is not sure about how to write it.

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+## 滚动升级 TiDB 集群组件
+
+> **注意：**
+>
+> 为优化 TiDB 集群组件的运维管理，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行了调整。与之前版本相比，滚动升级 TiDB 3.0 版本集群组件的操作略有不同，注意升级前后 `process_supervision` 参数配置须保持一致。
+
+如果 `process_supervision` 变量使用默认的 `systemd` 参数，则通过 `excessive_rolling_update.yml` 滚动升级 TiDB 集群。  
+```
```
Comment (liubo0127) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion

```
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+$ ansible-playbook local_prepare.yml
+```
+
+## 滚动升级 TiDB 集群组件
+
+> **注意：**
+>
+> 为优化 TiDB 集群组件的运维管理，TiDB 3.0 版本对 `systemd` 模式下的 `PD service` 名称进行了调整。与之前版本相比，滚动升级 TiDB 3.0 版本集群组件的操作略有不同，注意升级前后 `process_supervision` 参数配置须保持一致。
+
+如果 `process_supervision` 变量使用默认的 `systemd` 参数，则通过 `excessive_rolling_update.yml` 滚动升级 TiDB 集群。
```
Comment (liubo0127) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
如果 `process_supervision` 变量使用默认的 `systemd` 参数，则通过 `excessive_rolling_update.yml` 滚动升级 TiDB 集群。  

```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+
+## 在中控机器上下载 TiDB-Ansible
+
+以 `tidb` 用户登录中控机并进入 `/home/tidb` 目录，备份 TiDB 2.0 版本或 TiDB 2.1 版本的 tidb-ansible 文件夹：
+
+```
+$ mv tidb-ansible tidb-ansible-bak
+```
+
+下载最新 tidb-ansible v3.0.0 或者 release-3.0 [**分支**](/op-guide/ansible-deployment.md#在中控机器上下载-tidb-ansible)，默认的文件夹名称为 `tidb-ansible`。
```
Comment (liubo0127) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
下载 TiDB 3.0 版本对应 tag 的 tidb-ansible  [**下载 TiDB-Ansible**](/op-guide/ansible-deployment.md#在中控机器上下载-tidb-ansible)，默认的文件夹名称为 `tidb-ansible`。
```

File: op-guide/tidb-v3.0-upgrade-guide.md (hunk 14 above)
Comment (liubo0127) [reply op-guide/tidb-v3.0-upgrade-guide.md]: Suggest change to 'git clone -b $tag https://github.com/pingcap/tidb-ansible.git'

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+category: deployment
+---
+
+# TiDB 3.0 升级操作指南
+
+本文档适用于从 TiDB 2.0 版本（v2.0.1 及之后版本）或 TiDB 2.1 RC 版本升级到 TiDB 3.0 GA 版本。TiDB 3.0 版本兼容 [Kafka 版本的 TiDB-Binlog](/tools/tidb-binlog-kafka.md) 以及[集群模式的 TiDB-Binlog](/tools/tidb-binlog-cluster.md)。
+
+## 升级兼容性说明
+
+- loading
```
Comment (liubo0127) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
- 不支持在升级后回退至 2.1.x 或更旧版本
- 从 2.0.6 之前的版本升级到 3.0 之前，需要确认集群中是否存在正在运行中的 DDL 操作，特别是耗时的 `Add Index` 操作，等 DDL 操作完成后再执行升级操作
- 2.1 及之后版本启用了并行 DDL，早于 2.0.1 版本的集群，无法滚动升级到 3.0，可以选择下面两种方案：
    - 停机升级，直接从早于 2.0.1 的 TiDB 版本升级到 3.0
    - 先滚动升级到 2.0.1 或者之后的 2.0.x 版本，再滚动升级到 3.0 版本
```

File: op-guide/tidb-v3.0-upgrade-guide.md [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+
+## 注意事项
+
+- loading
```
Comment (liubo0127) [inline op-guide/tidb-v3.0-upgrade-guide.md]:
```suggestion
在升级的过程中不要执行 DDL 请求，否则可能会出现行为未定义的问题。
```

**Comment by lilin90** [comment]
@liubo0127 PTAL again~ Thanks!

**Pull request merged**